dali/python/nvidia/dali/plugin/pytorch/__init__.py

```diff
@@ -224,7 +224,10 @@ def __init__(
                 # here we should set if to False again
                 self._ever_consumed = False
             except StopIteration:
-                assert False, (
+                # This case might not be an error if we're iterating over pipeline that is
+                # currently at the end of epoch, for example because it was restored from
+                # checkpoint.
+                assert any(not p._first_iter for p in self._pipes), (
```

Comment:
I know that this already exists here, but throwing errors via triggering asserts is not a proper error checking. We should check the condition and throw the exceptions.

One can disable asserts in their code: https://docs.python.org/3/using/cmdline.html#cmdoption-O

Reply:
Changed to conditional `RuntimeError` in https://github.com/NVIDIA/DALI/pull/5213/commits/f2c648ca31b7414efc8ca95eb71c82109c0c84d6